Sequence refactor

jzaremba opened the pull request:
The goal of this refactorization is to only load data as needed, which removes unnecessary data transfer when datasets are accessed remotely. There should be no noticeable difference with local datasets, but if you don't actually need to iterate over the sequences of an ImagingDatasets, load times should be much faster.

Resolves Issue #104 

Major change:
- sequence data is pulled out of dataset.pkl in to sequences.pkl
- sequences, frame_shape, num_frames, num_sequences are now all properties instead of attributes and are initialized at load if stored in dataset.pkl (see below), or calculated as needed.

Minor changes:
- num_sequences and frame_shape are now saved in dataset.pkl
- time_averages is now saved in the object, so you do not need to re-read the time_averages.pkl file every time you call ImagingDataset.time_averages

Also:
- When prompting to overwrite a SIMA directory, the directory is now printed with the prompt
- A few bug fixes for ellipse ROIs in ROI_Buddy
- ROI_Buddy code cleanup

NOTE: Currently an ImportError is explicitly thrown if you try to load a 1.0.0.dev dataset, not sure if there is a better way to handle this.

I have a convert script that will convert SIMA 1.0.0.dev sets to this new format. I'm not going to put it in the repo, but I can pass it around if needed.

pkaifosh commented on sima/imaging.py:
```diff
@@ -102,27 +102,27 @@ def __init__(self, sequences, savedir, channel_names=None,
                 data = pickle.load(f)
-            self.sequences = [unpack(s) for s in data.pop('sequences')]
+            if 'sequences' in data:
+                raise ImportError('Old version')
```
Let's be a bit more specific about which version would have triggered this error (i.e. 1.0.0-dev?). Then, in some later version we will know whether it is likely safe to remove this check.

pkaifosh commented:
Looks great. Please go ahead and merge it in once the above issue is addressed and if you've tested it by creating, saving, and loading some datasets.

jzaremba commented:
Updated the ImportError message and added a comment to suggest the check get removed in a future SIMA version.

I've converted some of my own sets and worked with them successfully. 0.x sets still convert fine. Once all the tests pass again I'll pull it in to master.
